spec/factories/voucher_factory.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :voucher do
+    enterprise { create(:distributor_enterprise) }
+    amount { rand(200.0) }
+  end
```

Comment:
Great to have a factory. When I tried it out I noticed that enterprise is not a required field. We still have an old Rails config:

https://github.com/openfoodfoundation/openfoodnetwork/blob/482d438ec93f91bcc4f6aafa307fdfa2cffeb4d9/config/application.rb#L229

So we need to add `optional: false` to the model, I think.

Also, when you just `build(:voucher)` then it *creates* an enterprise unnecessarily. It would be better to use `build` here because it's much faster when you don't need to persist.

I personally don't like random test data. It can make tests flaky and they are more difficult to debug when they are not deterministic. Sometimes randomness can reveal bugs and therefore it's a trade-off but I prefer deterministic tests.

Reply:
@mkllnk I can see that the belongs_to required issue is resolved now, but I also think it's worth removing the random value. 

Although it could potentially help uncover new cases it would be random and suck up a lot of tech debt time with perhaps no benefit. So I don't think we should set this precedent, do you agree if I change this to a value like $15?